libcontainer/cgroups/fs/fs.go

```diff
@@ -308,6 +308,7 @@ func (m *manager) Set(container *configs.Config) error {
 		}
 	}
 
+	m.cgroups.Resources = r
```

Comment:
I see this has not been consistent between the managers. My only concern with this is that this makes it harder to understand what runnning `Set(nil)` will result in.

As far as I understand the libcontainer api (and oci in general), resources omitted in a resource "update" will be kept as before (eg. the cgroup files are not written). This will just "wipe" old state with the new one, afaik.

Doing something like

```go
mgr := fs.NewManager(fullContainerState)
mgr.Set(memory: 10Gib)           // will ignore everything else, but devices need SkipDevices
mgr.Set(cpu_shares: 100MiB)  // will ignore everything else, but devices need SkipDevices
mgr.Set(nil) // what do we want this to do?
```

has been a bit ambiguous, and I think all will benefit from explicitly stating how this api _should_ and _can_ be used. Constantly doing a full `Set` of all cgroups when dynamically changing cpusets or cfs quota should probably not have to involve writing to _all_ the cgroup files. I would like a way of setting separate things, but saying no to that is ok, as long as all users understand how it works.

Afaik.  the runc cli will write all cgroup files on a simple `runc update`, even though it not always is what we want.

As a background: k8s keep updating cpuset for all containers quite often, creating some extra load when all the cgroup controllers are updated....

Reply:
I agree with your analysis. What I was thinking of, as a next step, is for Set to compare the current (saved) and new (passed on in) values, and only set those that differ. Alternatively, set those that are not nil, and modify the internal state accordingly.

In any case, this is the next step -- for now I just want to make sure that Set argument is what it should be -- resources, not the whole cgroup config.

Reply:
Looks like we agree then. Sounds like a good plan!

I really like this change, so it looks good to me.